uaclient/util.py

```diff
@@ -416,16 +418,27 @@ def _subp(
     @raises subprocess.TimeoutError when timeout specified and the command
         exceeds that number of seconds.
     """
+    if data is None:
+        # using devnull assures any reads get null, rather
+        # than possibly waiting on input.
+        stdin = open(os.devnull)
```

Comment:
Can https://docs.python.org/3.4/library/subprocess.html#subprocess.DEVNULL be used here?

Reply:
yep and it should. Disregarded this comment because we dropped that functionality from this branch. But will remember that in the future.